### PR TITLE
Code Cleanup

### PR DESCRIPTION
**Pull request by pwpiwi**
- deduplicate crapto1 library
- deduplicate mfkey32, mfkey64 and nonce2key
- replace -D__USE_MINGW_ANSI_STDIO=1 by -D_ISOC99_SOURCE
- replace deprecated #define WIN32 by #define _WIN32
- remove armsrc/stdint.h

### CODE REVIEW

**Comment by marshmellow42** [comment]
i like it...  Good Job PIWI.  it compiles well in multiple environments, i will attempt to test some functionalities.

**Comment by pwpiwi** [comment]
It only looks so monster like because many double files are being deleted. If I would separate the last three bullets, the remaining PR would still be "monster".

**Comment by pwpiwi** [comment]
BTW: This change was initiated by post http://www.proxmark.org/forum/viewtopic.php?id=2771

File: client/Makefile [hunk 1]
```diff
@@ -9,17 +9,16 @@ include ../common/Makefile.common
...
 LDLIBS = -L/opt/local/lib -L/usr/local/lib -lreadline -lpthread -lm
 LUALIB = ../liblua/liblua.a
 LDFLAGS = $(COMMON_FLAGS)
-CFLAGS = -std=c99 -I. -I../include -I../common -I../zlib -I/opt/local/include -I../liblua -Wall $(COMMON_FLAGS) -g -O4
+CFLAGS = -std=c99 -D_ISOC99_SOURCE -I. -I../include -I../common -I../tools -I../zlib -I/opt/local/include -I../liblua -Wall $(COMMON_FLAGS) -g -O4
 LUAPLATFORM = generic
 
 ifneq (,$(findstring MINGW,$(platform)))
-    CFLAGS += -D__USE_MINGW_ANSI_STDIO=1
     CXXFLAGS = -I$(QTDIR)/include -I$(QTDIR)/include/QtCore -I$(QTDIR)/include/QtGui
```
Comment (marshmellow42) [inline client/Makefile]: without this one environment still complains about unknown print characters
`flasher.c: In function 'cmd_debug':
flasher.c:27:3: warning: unknown conversion type character 'z' in format [-Wformat=]
   printf("UsbCommand length[len=%zd]\n",sizeof(UsbCommand));
   ^
flasher.c:27:3: warning: too many arguments for format [-Wformat-extra-args]
flasher.c:28:3: warning: unknown conversion type character 'z' in format [-Wformat=]
   printf("  cmd[len=%zd]: %016"llx"\n",sizeof(UC->cmd),UC->cmd);
   ^`

**Comment by marshmellow42** [comment]
i also see this warning:
cmdhfmf.c: In function 'CmdHF14AMfCLoad':
cmdhfmf.c:1542: warning: 'flags' may be used uninitialized in this function

**Comment by marshmellow42** [comment]
these warnings also appear to be new:
arm-none-eabi-objcopy -F elf32-littlearm --remove-section .fpgaimage obj/fullimage.elf obj/osimage.elf
c:\devkitPro\devkitARM\bin\arm-none-eabi-objcopy.exe: obj/fullimage.elf: warning: Empty loadable segment detected, is this intentional ?

arm-none-eabi-objcopy -Osrec --srec-forceS3 --strip-debug --no-change-warnings --change-addresses -0x100000 --change-start 0 --change-section-address .bss+0 --change-section-address .data+0 --change-section-address .commonarea+0 obj/osimage.elf obj/osimage.s19
arm-none-eabi-objcopy -F elf32-littlearm --only-section .fpgaimage obj/fullimage.elf obj/fpgaimage.elf
c:\devkitPro\devkitARM\bin\arm-none-eabi-objcopy.exe: obj/fullimage.elf: warning: Empty loadable segment detected, is this intentional ?

c:\devkitPro\devkitARM\bin\arm-none-eabi-objcopy.exe: obj/fullimage.elf: warning: Empty loadable segment detected, is this intentional ?

c:\devkitPro\devkitARM\bin\arm-none-eabi-objcopy.exe: obj/fullimage.elf: warning: Empty loadable segment detected, is this intentional ?

File: client/Makefile (hunk 1 above)
Comment (pwpiwi) [inline client/Makefile]: This is for mingw? Does your machine compile with -D_ISOC99_SOURCE?
```
make[2]: Leaving directory `/home/piwi/github/proxmark3/liblua'
gcc -std=c99 -D_ISOC99_SOURCE -I. -I../include -I../common -I../tools -I../zlib -I/opt/local/include -I../liblua -Wall  -g -O4 -DHAVE_GUI -DZ_SOLO -DZ_PREFIX -DNO_GZIP -DZLIB_PM3_TUNED  -c -o obj/flasher.o flasher.c
g++ -I/local32/qt4/include -I/local32/qt4/include/QtCore -I/local32/qt4/include/QtGui obj/flash.o obj/flasher.o obj/uart.o obj/util.o obj/sleep.o -L/opt/local/lib -L/usr/local/lib -lreadline -lpthread -lm -o flasher
make[1]: Leaving directory `/home/piwi/github/proxmark3/client'
piwi@miraculix ~/github/proxmark3$
```

Comment (marshmellow42) [inline client/Makefile]: mingw correct.  that compiler flag appears to be missing.  where do i set it?

**Comment by pwpiwi** [comment]
I forgot to add some files to the PR. Please try again (best to clone from scratch, just in case you have some old files lying around).

File: client/Makefile (hunk 1 above)
Comment (marshmellow42) [inline client/Makefile]: wow...  please ignore this error...  i was compiling your master branch not the cleanup branch when i got this one..  sorry

**Comment by iceman1001** [comment]
I still standby my review,  _divide and make minor PRs_

**Comment by marshmellow42** [comment]
once i figured out my git issues...  and piwi's latest changes, i have no more issues so far with it on multiple mingw environments.

**Comment by marshmellow42** [comment]
most of the changes are re-arranging the order of functions and changing headers files as needed.  (which has been overdue.) 

i have now verified mifare crypto functions are still working well.  (including reader attack, sniff and reading/writing of mifare tags, nested attack...)

**Comment by marshmellow42** [comment]
it appears all the below file changes are minor header changes or single fixes, and might be able to be separated out.
```
armsrc/BigBuf.c
armsrc/appmain.c
armsrc/apps.h
armsrc/des.c
armsrc/epa.c
armsrc/optimized_cipher.c
armsrc/pcf7931.c
armsrc/printf.c
armsrc/string.h
client/cmdhf.c
client/cmdhfmfu.c
clinet/cmdmain.h
client/whereami.c
common/iso14443crc.h
include/hitag2.h
include/hitagS.h
```
though i see nothing wrong with them.

i do appreciate iceman's desire to keep changes small and easily track-able.  i would still tend to vote to accept this pull, especially based on piwi's history of quickly resolving any issues that arise within his code changes.  (especially since i'd rather piwi continue using his time patching..)  :) 

not that my opinion really matters...  just thought i'd throw in anyway...  Keep up the great work everyone!

**Comment by pwpiwi** [comment]
Thanks @marshmellow42 for listing the easy ones. This will at least bring the number of affected files down. I will close this PR and raise a few more...

**Pull request closed without merging**